firefox-ios/Client/Frontend/Translations/TranslationsMiddleware.swift

```
@@ -104,21 +111,31 @@ final class TranslationsMiddleware {
         store.dispatch(toolbarAction)
     }
 
-    // TODO: FXIOS-13844 - Check if we can translate a page based on certain eligibility
-    @MainActor
+    /// Checks whether the current page in the active tab is eligible for translation,
+    /// and if so, dispatches a toolbar action to update the translation state.
     private func checkTranslationsAreEligible(for action: ToolbarAction) {
-        // We dispatch an action for now, but eventually we want to inject a script
-        // to check if the page language differs from our locale language.
-        guard action.translationConfiguration?.canTranslate == true else { return }
-        let toolbarAction = ToolbarAction(
-            translationConfiguration: TranslationConfiguration(
-                prefs: profile.prefs,
-                state: .inactive
-            ),
-            windowUUID: action.windowUUID,
-            actionType: ToolbarActionType.receivedTranslationLanguage
-        )
-        store.dispatch(toolbarAction)
+        Task { @MainActor in
+            guard action.translationConfiguration?.canTranslate == true else { return }
+
+            guard let selectedTab = self.windowManager.tabManager(for: action.windowUUID).selectedTab,
+                  let webView = selectedTab.webView
+            else { return }
+
+            let languageSampleSource = WebViewLanguageSampleSource(webView: webView)
+            let pageLanguage = try await languageDetector.detectLanguage(from: languageSampleSource)
```

When I tried running locally, I'm not getting the translate button to appear for French Wikipedia pages. 🤔  I see this in the console:
```
WebContent[91146] Unable to filter tracking query parameters (missing data)
WebContent[91146] 0x106809e08 - [webPageID=8] WebPage::runJavaScriptInFrameInScriptWorld: Request to run JavaScript failed with error TypeError: undefined is not an object (evaluating 'window.__firefox__.Translations')
```
Do you know why this might be?

P.S. This `try` is not wrapped in a do-catch so the error is basically getting swallowed and ignored by the Task. We should probably try to handle it properly though with a `catch`. Here is the error I'm seeing if you catch it:
```
Error Domain=WKErrorDomain Code=4 "A JavaScript exception occurred" UserInfo={WKJavaScriptExceptionLineNumber=0, WKJavaScriptExceptionMessage=TypeError: undefined is not an object (evaluating 'window.__firefox__.Translations'), WKJavaScriptExceptionColumnNumber=0, NSLocalizedDescription=A JavaScript exception occurred}
```

---

Ohhh gochya!! Ok. 😁 👍 Your JavaScripting is like dark wizard magic to me heh. 👀  Can you CC me on that work next PR just so I can see what sort of stuff you're doing @issammani ?

P.S. My comment about catching the errors thrown in a Task still stands tho @cyndichin !